### PR TITLE
ROCANA-6502 Add support for collecting individual network connections (Linux)

### PR DESCRIPTION
**Pull request by actgardner**
Adds per-connection metrics for TCP, UDP and raw sockets over IPv4 and v6. We don't capture everything, just the hosts, ports, queue depth and connection state.

### CODE REVIEW

File: sigar_linux.go [hunk 1]
```diff
@@ -216,6 +217,146 @@ func (self *NetIfaceList) Get() error {
...
+		}
+
+		connList = append(connList, conn)
+		return true
+	})
+	return connList, err
+}
+
+/* Decode an IP:port pair, with either a 16-byte or 4-byte IP as specified by lenBytes
+   encoded in network order as hexidecimal characters ex. 0100007F -> 127.0.0.1 */
```
Comment (adamlamar) [inline sigar_linux.go]: This comment seems off as network order would be big endian, e.g. `7F000001` for `127.0.0.1`.

Comment (actgardner) [inline sigar_linux.go]: Yeah, I didn't bother checking which one was "network". It's little-endian like the example - gonna fix the comment.

File: sigar_linux.go [hunk 2]
```diff
@@ -216,6 +217,146 @@ func (self *NetIfaceList) Get() error {
...
+	var port int64
+	var err error
+
+	port, err = strconv.ParseInt(parts[1], 16, 64)
+	if err != nil {
+		return nil, 0, fmt.Errorf("Unable to parse port, %v - %v", parts[1], err)
+	}
+
+	ip := make([]byte, lenBytes)
+	// For IPv6 the words are in order, but the bytes of the words are little-endian
```
Comment (adamlamar) [inline sigar_linux.go]: Are you sure these are always little endian? My guess is they are host byte order, so they could be big endian on architectures other than ia32/amd64.

Comment (actgardner) [inline sigar_linux.go]: I dug around in the kernel source and couldn't find a definite answer for this. We might have to punt until we have to support Linux on a big-endian system - there may be other complications moving the agent to POWER or ARM systems.

**Comment by actgardner** [comment]
@adamlamar I fixed the comment to clarify that the byte order is unknown on big-endian systems.

**Comment by adamlamar** [comment]
LGTM!

**Pull request closed without merging**